gst/gst_message_constructors.go

```diff
@@ -8,20 +8,24 @@ import (
 	"github.com/go-gst/go-glib/glib"
 )
 
-func getMessageSourceObj(src interface{}) *C.GstObject {
-	usafe, ok := src.(interface{ Unsafe() unsafe.Pointer })
-	if !ok {
+type Unsafer interface {
+	Unsafe() unsafe.Pointer
+}
```

Comment:
if we extract this, we should name it better and document it. Maybe `EmbedsGstObject` ?

Also maybe we could use the `GstObject` function that is present on every struct embedding `gst.Object`. This is a little different from the current code, but only if you are writing your own bindings and not extending `gst.Object`.

https://github.com/go-gst/go-gst/blob/main/gst/gst_object.go#L30

What do you think?

Reply:
That interface name follows an offical Go convention on one method interface naming.
https://go.dev/doc/effective_go#interface-names

I see that you are thinking forward on how to make it better - and I understand that. 
But currently I find a usability issue in that it is checking type erased arguments if they do contain Unsafe() method where it is clearly against Go interfacing feature and common sense.
Go's idiomatic way is in that If you need to set a constraint for the function argument to have a specific method on it - you must use interface with defined behaivor. 
That is beneficial for both the end-user which won't be guessing if this type is suitable in here and beneficial for the developers to remove unneeded casting checks. 

I do agree that interface method checking thing is indeed a cool trick in other places where marshaling and OOP like deriving are performed but those are in very far away from a user places so only developers have to face it.

On the contrary messages related api is facing user and I suggest making it a little more concise of what it expects as argument instead of type erased interface{}.

I would also suggest to create a separate issue topic about how to unify that raw pointer retrieval thing to make it more clear because I think it cannot be done in a fast and simple way and needs deeper discussions.

https://www.alexedwards.net/blog/interfaces-explained

Reply:
> But currently I find a usability issue in that it is checking type erased arguments if they do contain Unsafe() method where it is clearly against Go interfacing feature and common sense.

I see your point, the current implementation is totally misleading. We should change it. If we change, why not do the follwowing:

Since the message always requires a (nullable) pointer to a `GstObject`, I think it would be better to use an interface similar to:

```go
type EmbedsGstObject interface {
  GstObject() *C.GstObject
}
```

This interface is implemented by every struct in `go-gst` that extends `GstObject`: `Pipeline`, `Bin`, `Element`, `Pads` etc.

Since it requires the C bindings, it is also not implementable by user code (Which I think is a good thing, users should embed `gst.Object` if they extend it).

Reply:
That looks way better than Unsafe. What about GstObjectHolder as a name?

Reply:
Yeah something like that. Maybe `GstObjectChild`? The thing is that in C this tries to be an object oriented class hierarchy, which does not exist in go, so the interface name has to be a bit unconventional I think.

Feel free to settle on a name you find fitting, as long as it is documented and somewhat clear to the user, it is okay.

Reply:
I think that a *Child is not really represents what is reads due to in golang there is no inheritance and it is more like a struct that in some way has access to a pointer to the gst object glib's derived instance.

I added a little comment for that interface.